src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -426,100 +432,103 @@ public void acceptRecord(final SAMRecord record) {
+                    for (final Interval bait : probes) {
+                        for (final AlignmentBlock block : record.getAlignmentBlocks()) {
+                            final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
 
-            // Find the target overlaps
-            if (targets != null && !targets.isEmpty()) {
-                for (final Interval target : targets) {
-                    final Coverage coverage = this.coverageByTarget.get(target);
-
-                    for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
-                        final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-                        for (int pos=block.getReferenceStart(), readPos = block.getReadStart();
-                             pos<=end && pos >= target.getStart() && pos <= target.getEnd() && minimumBaseQuality <= baseQualities[readPos-1];
-                             ++pos, ++readPos) {
-                            ++this.metrics.ON_TARGET_BASES;
-                            if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
-                            coverage.addBase(pos - target.getStart());
+                            for (int pos=block.getReferenceStart(); pos<=end; ++pos) {
+                                if (pos >= bait.getStart() && pos <= bait.getEnd()) ++onBaitBases;
+                            }
                         }
                     }
+
+                    this.metrics.ON_PROBE_BASES   += onBaitBases;
+                    this.metrics.NEAR_PROBE_BASES += (mappedBases - onBaitBases);
+                }
+                else {
+                    this.metrics.OFF_PROBE_BASES += mappedBases;
                 }
             }
 
-            // Now do the bait overlaps
-            int mappedBases = 0;
-            for (final AlignmentBlock block : rec.getAlignmentBlocks()) mappedBases += TargetMetricsCollector.getNumBasesPassingMinimumBaseQuality(rec, block, minimumBaseQuality);
-            int onBaitBases = 0;
-
-            if (probes != null && !probes.isEmpty()) {
-                for (final Interval bait : probes) {
-                    for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
-                        final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-                        for (int pos=block.getReferenceStart(), readPos = block.getReadStart();
-                             pos<=end && pos >= bait.getStart() && pos <= bait.getEnd() && minimumBaseQuality <= baseQualities[readPos-1];
-                             ++pos, ++readPos) {
-                            ++onBaitBases;
+            ///////////////////////////////////////////////////////////////////
+            // And lastly, ignore reads falling below the mapq threshold
+            ///////////////////////////////////////////////////////////////////
+            if (this.mapQFilter.filterOut(record)) return;
```

Comment:
This was very much a judgement call, and open to debate.  The way I see HsMetrics is that there are two classes of metrics in here:
1. _Target_ related metrics are for assessing how useful the data will be in downstream applications - i.e. variant calling
2. _Bait_ metrics are for assessing how well the wetlab assay worked.

The `ON/NEAR/OFF_PROBE_BASES` fall into (2) and are the metrics from which `PCT_SELECTED` and `FOLD_ENRICHMENT` are derived.

So while it's clear to me that quality cutoffs, and overlaps etc. should be discounted from (1), I feel like doing so for (2) can skew the picture of how the assay is working.  E.g. one change I reverted was that bases that were over a bait, but of insufficient base quality, were being counted as "off bait", which meant that when quality was lower, %selected among other metrics dropped off.  That definitely seemed wrong.  I feel similarly about overlap clipping.  And to a certain degree mapping quality.